rs/execution_environment/src/canister_manager/tests.rs

```diff
@@ -8103,118 +8103,91 @@ fn helper_update_settings_updates_hook_status(
-                settings = settings.with_wasm_memory_threshold(updated_wasm_memory_threshold);
-            }
+    if updated_memory_state.wasm_memory_threshold != initial_memory_state.wasm_memory_threshold {
+        if let Some(updated_wasm_memory_threshold) = updated_memory_state.wasm_memory_threshold {
+            updated_settings =
+                updated_settings.with_wasm_memory_threshold(updated_wasm_memory_threshold.get());
         }
+    }
 
-        if updated_memory_state.wasm_memory_limit != initial_memory_state.wasm_memory_limit {
-            if let Some(updated_wasm_memory_limit) = updated_memory_state.wasm_memory_limit {
-                settings = settings.with_wasm_memory_limit(updated_wasm_memory_limit);
-            }
+    if updated_memory_state.wasm_memory_limit != initial_memory_state.wasm_memory_limit {
+        if let Some(updated_wasm_memory_limit) = updated_memory_state.wasm_memory_limit {
+            updated_settings =
+                updated_settings.with_wasm_memory_limit(updated_wasm_memory_limit.get());
         }
+    }
 
-        if updated_memory_state.memory_allocation != initial_memory_state.memory_allocation {
-            if let Some(updated_memory_allocation) = updated_memory_state.memory_allocation {
-                settings = settings.with_memory_allocation(updated_memory_allocation);
-            }
+    if updated_memory_state.memory_allocation != initial_memory_state.memory_allocation {
+        if let Some(MemoryAllocation::Reserved(updated_memory_allocation)) =
+            updated_memory_state.memory_allocation
+        {
+            updated_settings =
+                updated_settings.with_memory_allocation(updated_memory_allocation.get());
         }
+    }
 
-        let canister = state.canister_state_mut(&canister_id).unwrap();
+    let payload = UpdateSettingsArgs {
+        canister_id: canister_id.get(),
+        settings: updated_settings.build(),
+        sender_canister_version: None,
+    }
+    .encode();
 
-        assert!(canister_manager
-            .update_settings(
-                Time::from_nanos_since_unix_epoch(777),
-                canister_change_origin_from_principal(&sender),
-                settings.build(),
-                canister,
-                &mut round_limits,
-                ResourceSaturation::default(),
-                SMALL_APP_SUBNET_MAX_SIZE,
-            )
-            .is_ok());
+    test.subnet_message(Method::UpdateSettings, payload)
+        .unwrap();
 
-        assert_eq!(
-            state
-                .canister_state_mut(&canister_id)
-                .unwrap()
-                .system_state
-                .task_queue
-                .peek_hook_status(),
-            updated_memory_state.hook_status
-        );
-    })
+    assert_eq!(
+        test.canister_state_mut(canister_id)
+            .system_state
+            .task_queue
+            .peek_hook_status(),
+        updated_memory_state.hook_status
+    );
 }
```

Comment:
This change only refactors helper test to become ExecutionEnvironmentTest so that we can test the execution of `fn execute_subnet_message` at the end of which execution hook status is updated.